pyiceberg/table/snapshots.py

```diff
@@ -116,3 +144,199 @@ class MetadataLogEntry(IcebergBaseModel):
 class SnapshotLogEntry(IcebergBaseModel):
     snapshot_id: int = Field(alias="snapshot-id")
     timestamp_ms: int = Field(alias="timestamp-ms")
+
+
+class SnapshotSummaryCollector:
+    added_size: int
+    removed_size: int
+    added_files: int
+    removed_files: int
+    added_eq_delete_files: int
+    removed_eq_delete_files: int
+    added_pos_delete_files: int
+    removed_pos_delete_files: int
+    added_delete_files: int
+    removed_delete_files: int
+    added_records: int
+    deleted_records: int
+    added_pos_deletes: int
+    removed_pos_deletes: int
+    added_eq_deletes: int
+    removed_eq_deletes: int
+
```

Comment:
I was hoping to get some insights from @rdblue on this one.

Reply:
The flag is needed because we don't have the correct counts for the [eq and pos deletes in delete manifests](https://github.com/apache/iceberg/pull/1367/files#r500492383). I don't think that we need to add whole manifests in Python so I'd skip it.